helix/appendix/alternate-visual-studio-structures.rst

```diff
@@ -0,0 +1,231 @@
+
+    The TDS version of Helix Basic Company uses a single TDS project for
+    file copy / packaging, and additional TDS projects as-needed within each module.
+    It also only has test projects for modules with testable logic.
+
+    .. figure:: _static/basic-company-projects.png
+        :scale: 75%
+
+Are there inefficiencies in your build scripts?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Previous versions of Sitecore Habitat included gulp scripts which built each
+Visual Studio project individually, rather than the entire solution at once.
+This could lead to much slower builds as project dependencies could be built
+more than once. Ensure your build scripts target the VS Solution or a single
+"root-level" project.
+
+.. admonition:: Sitecore Habitat
+
+    Newer versions of `Sitecore Habitat <https://github.com/Sitecore/Habitat>`__
+    have an updated ``gulpfile`` which targets the entirety of the Visual Studio
+    Solution for builds.
+
+    .. code-block:: js
+
+        gulp.task("Publish-All-Projects",
+            function(callback) {
+                var msbuild = new _msbuild(callback);
+                msbuild.solutionName = config.solutionName + '.sln';
+                var overrideParams = [];
+                overrideParams.push('/p:Configuration=' + config.buildConfiguration);
+                overrideParams.push('/p:DeployOnBuild=true');
+                overrideParams.push('/p:DeployDefaultTarget=WebPublish');
+                overrideParams.push('/p:WebPublishMethod=FileSystem');
+                overrideParams.push('/p:DeleteExistingFiles=false');
+                overrideParams.push('/p:publishUrl=' + config.websiteRoot);
+                overrideParams.push('/m');
+                overrideParams.push('/restore');
+                msbuild.config('overrideParams', overrideParams);
+                msbuild.config('version', config.buildToolsVersion);
+                msbuild.build();
+            });
+
+    You can alternatively use Sitecore TDS or Helix Publishing Pipeline
+    for local deployments, as seen in the new Sitecore Helix Examples.
+
+
+Are there other tooling improvements which can help?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If your primary concern is the overhead of creating projects which follow
+the Sitecore Helix conventions (naming, filesystem structure, etc.), then
+you might consider some of the community tools which make this easier by
+providing project "templates" which save you some clicks. Take care when
+using these however that you remove any generated projects
+`which the module does not need <#does-your-solution-contain-projects-you-do-not-need>`__.
+
+To improve the performance of Visual Studio and reduce build times, you can
+also consider using the `Filtered Solutions <https://docs.microsoft.com/en-us/visualstudio/ide/filtered-solutions?view=vs-2019>`__
+feature of Visual Studio 2019 to reduce the number of projects open when working
+on a particular module.
```

Comment:
Have seen other customers use multiple solutions for different teams -- ie: developers use a solution with Foundation modules as nuget references, and only smaller subsets of Feature and Project modules relevant to a tenant set. Architects would have a full set solution with all modules.

Reply:
Isn't the challenge there ensuring that each developer is consuming/building a fully integrated solution? Seems like a big benefit of the solution filters that you don't lose context of the larger solution.